### PR TITLE
add placeholders for resources that are not present

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1712/
Test PASSed.

File: internal/charmstore/resources_test.go [hunk 1]
```diff
@@ -170,11 +170,16 @@ func (s *resourceSuite) TestListResourcesResourceNotFound(c *gc.C) {
 	sortResources(expected)
 	err := store.PublishResources(entity, channel, resourceRevisions(expected[1:]))
```
Comment (rogpeppe) [inline internal/charmstore/resources_test.go]: I don't think PublishResources should allow publishing without all the resources
that are needed by the charm, so I'd write this test a bit differently.

**Comment by rogpeppe** [comment]
Looks reasonable except for the fact that we shouldn't
be allowed to do what the test is doing here.
Land this then fix that if it's easier.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1720/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1721/
Test PASSed.

**Comment by rogpeppe** [comment]
LGTM thanks!

**Comment by urosj** [comment]
👍 However, have a bit of reservations how the whole experience would look like. If we can reserve a name for the resource but not require the resource to be uploaded, then it seems fine to return -1 for the revision.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1722/
Test PASSed.

**Pull request merged**